### ISSUE TITLE
Autoplay keeps trying to transition after the component has been unmounted

### ISSUE THREAD

**Issue by eparizzi** [leandrowd/react-responsive-carousel#151]
I have a custom component (let's call it `Container`) that has a carousel with `autoPlay=true` in it. Everything works well, but when I remove my `Container` component (and hence the carousel inside of it), I start to get errors in the console every amount of time defined in the `interval` property of the carousel.

The error is the following:
`Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Carousel component.`

**Comment by ptshih** [comment]
I also have this issue, haven't had time to fully debug it, but I think it has to do with the way the `setTimeout` timers are created and cleared...

**Comment by emzero** [comment]
@ptshih Yes, it's clear that the timer is not being cleared when the component is dismounted.

**Comment by leandrowd** [comment]
Hey guys, I won't have time to dedicate to this plugin for a while. Would
any of you be interested in contributing?

**Comment by Charliekenney23** [comment]
I'm trying to clear some issues on the repo with a big pull request, and I'm unable to reproduce this in my environment (using conditional rendering). Is there anything different about your implementation that I should look into?

This should be resolved in the callstack:
`componentWillUnmount()` ([Carousel.js:105](https://github.com/leandrowd/react-responsive-carousel/blob/master/src/components/Carousel.js#L105))
-->`this.destroyCarousel()` ([Carousel.js:106](https://github.com/leandrowd/react-responsive-carousel/blob/master/src/components/Carousel.js#L106))
---->`this.destroyAutoPlay()` ([Carousel.js:132](https://github.com/leandrowd/react-responsive-carousel/blob/master/src/components/Carousel.js#L132))
------>`this.clearAutoPlay()` ([Carousel.js:193](https://github.com/leandrowd/react-responsive-carousel/blob/master/src/components/Carousel.js#L193))
-------->`clearInterval(this.timer)` ([Carousel.js:198](https://github.com/leandrowd/react-responsive-carousel/blob/master/src/components/Carousel.js#L198))


My attempt at reproducing:
```js
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Carousel from './components/Carousel';

function _onChange() {
    console.log('onChange', arguments);
}

function _onClickItem() {
    console.log('onClickItem', arguments);
}

function _onClickThumb() {
    console.log('onClickThumb', arguments);
}

class DemoCarousel extends Component {
    render() {
        return (
            <Carousel
                showArrows
                emulateTouch
                infiniteLoop
                autoPlay
                interval={500}
                onChange={_onChange}
                onClickItem={_onClickItem}
                onThumbClick={_onClickThumb}
            >
                <div>
                    <img src="assets/1.jpeg" />
                    <p className="legend">Legend 1</p>
                </div>
                <div>
                    <img src="assets/2.jpeg" />
                    <p className="legend">Legend 2</p>
                </div>
                <div>
                    <img src="assets/3.jpeg" />
                    <p className="legend">Legend 3</p>
                </div>
                <div>
                    <img src="assets/4.jpeg" />
                    <p className="legend">Legend 4</p>
                </div>
                <div>
                    <img src="assets/5.jpeg" />
                    <p className="legend">Legend 5</p>
                </div>
                <div>
                    <img src="assets/6.jpeg" />
                    <p className="legend">Legend 6</p>
                </div>
            </Carousel>
        )
    }

    componentWillUnmount() {
        console.log('cya');
    }
}

class Demo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            show: true,
        }
    }

    render() {
        return (
            <div>
                { this.state.show ? <DemoCarousel/>: null }
                <button onClick={this.handleDisplayToggle}>
                    Hide me!
                </button>
            </div>
        )
    }

    handleDisplayToggle = () => {
        this.setState({
            show: !this.state.show
        })
    }
}

ReactDOM.render(<Demo />, document.querySelector('.demo-carousel'));
```

Thanks,

Charles

**Comment by leandrowd** [comment]
@emzero?

**Comment by devfive** [comment]
Hi,

I had similar error on my end. After debugging I noticed, that when swipe is done, `autoPlay` function is called, but it doesn't clear previous timer.

Regards

**Comment by leandrowd** [comment]
The PR from @devfive was published to npm (3.1.27).
